README.md

```diff
@@ -7,9 +7,85 @@ Table of Contents
 =================
 
 * [Name](#name)
+* [Synopsis](#synopsis)
 * [Description](#description)
 * [Copyright & License](#copyright--license)
 
+Synopsis
+========
+
+Basic usage
+
+```
+
+# set variable in ob file scope.
+foo = bar;  we create a variable foo which value is bar. variable in ob is string 
+replace , op will render variable like some template engine.
+
+# comment in ob file.
+#foo = bar; ob use # as comment syntax.
+
+# block in ob file.
+foo {
+    command1; 
+    command2; 
+}
+
+Block is basic execute unit.we use op compile out pl with make keyword to execute wanted unit.
+
+./openresty-test.pl make foo 
+
+# command
+All opsboy command will list below.
+
+# git
+git git@github.com location;  #git clone resource to location.
+
+# file
+file location; # check if file exist, if not create.
+
+# running
+running 'process matching part'; # check process is running.
+
+# dep
+dep block1 block2 block3; # execute depend block.
+
+# cwd
+ cwd location; # change working directory.
+
+# test
+test cmd; # if system(cmd) != 0 will not execute the block.
+
+# env
+env key value; #just working like export command;
+env PATH '~/git/lua-nginx-module/work/nginx/sbin:$PATH'; # we can use shell variable here.
+
+# always
+always; # exec blocking without conditions.
+
+# sh
+sh 'CMD'; # exec some shell command
+
+# yum
+yum 'PKG'; # yum install some package
+
+# debuginfo
+debuginfo 'kernel-`uname -r`'; #install debuginfo.
+
+# prog
+prog program; #check program can be execute.
+
+# fetch
+fetch source; # wget -N source.
```

Comment:
`wget` is an implementation detail which is subject to change in the future. We should document the functionality and behavior instead.